### PR TITLE
docs: Add advanced installation section for Private Clusters and APIServer Request Throttling

### PR DESCRIPTION
**Pull request by jonathan-innis**
Fixes #N/A 

**Description**

This adds a section to the Getting Started/Installation guide around installing Karpenter using a private cluster as well as installing Karpenter to prevent request throttling with FlowSchemas and PriorityLevelConfigurations.

There is a follow-up task here which involves adding the `spec.instanceProfile` suggestion to the `v0.32` and `docs` links when we do the `v0.32.2` release since that suggestion isn't valid now but will be once we do the patch release.

**How was this change tested?**

**Does this change impact docs?**
- [x] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [ ] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: website/content/en/preview/getting-started/getting-started-with-karpenter/_index.md [hunk 1]
```diff
@@ -145,19 +157,63 @@ The new stack has only one user, `admin`, and the password is stored in a secret
...
+
+{{% alert title="Note" color="primary" %}}
+
+Karpenter (controller and webhook deployment) container images must be in or copied to Amazon ECR private or to a another private registry accessible from inside the VPC. If these are not available from within the VPC, or from networks peered with the VPC, you will get Image pull errors when Kubernetes tries to pull these images from ECR public.
+
+{{% /alert %}}
+
+{{% alert title="Warning" color="warning" %}}
+
+There is currently no VPC private endpoint for the [Price List Query API](https://docs.aws.amazon.com/awsaccountbilling/latest/aboutv2/using-price-list-query-api.html). As a result, pricing data can go stale over time. By default, Karpenter ships a static price list that is updated when each binary is released.
```
Comment (bwagner5) [inline website/content/en/preview/getting-started/getting-started-with-karpenter/_index.md:199]: Need to set isolatedVPC: true in the helm chart too  https://github.com/aws/karpenter/blob/main/charts/karpenter/values.yaml#L190

File: website/content/en/docs/getting-started/getting-started-with-karpenter/_index.md [hunk 2]
```diff
@@ -145,19 +157,63 @@ The new stack has only one user, `admin`, and the password is stored in a secret
...
+caused by: Post "https://api.pricing.us-east-1.amazonaws.com/": dial tcp 52.94.231.236:443: i/o timeout, using existing pricing data from 2022-08-17T00:19:52Z  {"commit": "4b5f953"}
+```
+
+{{% /alert %}}
+
+### Preventing APIServer Request Throttling
+
+Kubernetes uses [FlowSchemas](https://kubernetes.io/docs/concepts/cluster-administration/flow-control/#flowschema) and [PriorityLevelConfigurations](https://kubernetes.io/docs/concepts/cluster-administration/flow-control/#prioritylevelconfiguration) to map calls to the API server into buckets which determine each user agent's throttling limits.
+
+By default, Karpenter is placed in the `workload-low` PriorityLevelConfiguration for all APIServer requests. This means that other components that make a high number of requests to the APIServer may affect the ability for Karpenter to make requests.
```
Comment (bwagner5) [inline website/content/en/docs/getting-started/getting-started-with-karpenter/_index.md:215]: is this only when Karpenter is not deployed in `kube-system`?

Comment (bwagner5) [reply website/content/en/docs/getting-started/getting-started-with-karpenter/_index.md:215]: Oh looks like only leaderelection is prioritized when in kube-system

File: website/content/en/preview/getting-started/getting-started-with-karpenter/_index.md (hunk 1 above)
Comment (jonathan-innis) [reply website/content/en/preview/getting-started/getting-started-with-karpenter/_index.md:199]: Ah, good callout 👍

File: website/content/en/docs/getting-started/getting-started-with-karpenter/_index.md (hunk 2 above)
Comment (jonathan-innis) [reply website/content/en/docs/getting-started/getting-started-with-karpenter/_index.md:215]: Both are prioritized. If you look at the FlowSchemas, you can see that there are mappings to leader-election for leader election requests and workload-high for all other requests with the `workload-leader-election` and `kube-system-service-accounts` FlowSchemas

File: website/content/en/preview/getting-started/getting-started-with-karpenter/_index.md [hunk 3]
```diff
@@ -145,19 +157,63 @@ The new stack has only one user, `admin`, and the password is stored in a secret
...
+You can optionally install Karpenter on a [private cluster](https://docs.aws.amazon.com/eks/latest/userguide/private-clusters.html#private-cluster-requirements) using the `eksctl` installation by setting `privateCluster.enabled` to true in your [ClusterConfig](https://eksctl.io/usage/eks-private-cluster/#eks-fully-private-cluster)
 
-### Delete the cluster
-To avoid additional charges, remove the demo infrastructure from your AWS account.
+```bash
+privateCluster:
+  enabled: true
+```
+
+Private clusters have no outbound access to the internet. This means that in order for Karpenter to reach out to the services that it needs to access, you need to enable specific VPC private endpoints. Below shows the endpoints that you need to enable to successfully run Karpenter in a private cluster:
```
[the remark this answers is not in the document]
Comment (jonathan-innis) [reply website/content/en/preview/getting-started/getting-started-with-karpenter/_index.md:173]: Good callout 👍

File: website/content/en/docs/getting-started/getting-started-with-karpenter/_index.md (hunk 2 above)
Comment (bwagner5) [reply website/content/en/docs/getting-started/getting-started-with-karpenter/_index.md:215]: Yeah, I meant if you don't have a FlowSchema at all and just put Karpenter in kube-system, leader election is prioritized but not other traffic.

File: website/content/en/docs/getting-started/getting-started-with-karpenter/_index.md [hunk 4]
```diff
@@ -44,7 +44,8 @@ authenticate properly by running `aws sts get-caller-identity`.
 After setting up the tools, set the Karpenter and Kubernetes version:
 
 ```bash
-export KARPENTER_VERSION=v0.32.1
```
Comment (bwagner5) [inline website/content/en/docs/getting-started/getting-started-with-karpenter/_index.md:47]: is this intentional to change?

Comment (jonathan-innis) [reply website/content/en/docs/getting-started/getting-started-with-karpenter/_index.md:47]: Unintentional. Good catch.

**Review by bwagner5 — DISMISSED** [review]
lgtm

**Review by bwagner5 — APPROVED** [review]
lgtm

**Comment by jonathan-innis** [comment]
This PR adds the FlowSchemas that were originally proposed inside of: https://github.com/aws/karpenter/issues/4168. This should help with apiserver contention when there are ill-behaving components that are making expensive requests within `workload-low`.

**Pull request merged**